include/mbedtls/ccm.h

```diff
@@ -143,8 +144,8 @@ int mbedtls_ccm_encrypt_and_tag( mbedtls_ccm_context *ctx, size_t length,
  * \param tag_len   The length of the tag in Bytes.
  *                  4, 6, 8, 10, 12, 14 or 16.
  *
- * \return          0 if successful and authenticated.
- * \return          #MBEDTLS_ERR_CCM_AUTH_FAILED if the tag does not match.
+ * \return          \c 0 on success.
```

Comment:
The previous wording “successful and authenticated” was awkward, but “authenticated” is useful information here. (Because there are APIs for CCM where “success” means “the calculation gave a result” and you need to check a separate output to determine whether the calculation gave the _expected_ result). How about:
```
\return \c 0 on success. This indicates that the message is authentic.
\return #MBEDTLS_ERR_CCM_AUTH_FAILED if the tag does not match.
\return A cipher-specific error code if the calculation failed.
```

Reply:
done (mbedtls_ccm_auth_decrypt)